velocity/src/main/java/us/myles/ViaVersion/VelocityPlugin.java

```diff
@@ -215,4 +229,40 @@ public JsonObject getDump() {
     public boolean isOldClientsAllowed() {
         return true;
     }
+
+    @Override
+    public java.util.logging.Logger getLogger() {
+        return logger;
+    }
+
+    @Override
+    public void onLoginSuccess(UserConnection connection) {
+        Objects.requireNonNull(connection, "connection is null!");
+        UUID id = connection.get(ProtocolInfo.class).getUuid();
+        connections.add(connection);
+        clients.put(id, connection);
+    }
+
+    @Override
+    public void onDisconnect(UserConnection connection) {
+        Objects.requireNonNull(connection, "connection is null!");
+        UUID id = connection.get(ProtocolInfo.class).getUuid();
+        connections.remove(connection);
+        clients.remove(id);
+    }
```

Comment:
Maybe it's worth having an abstractmanager instead of duplicating this code and maps every time